File: scripts/build-embed.py
"""Build offline embed scripts (no Node.js required)."""
import json
import pathlib

ROOT = pathlib.Path(__file__).resolve().parent.parent


def main() -> None:
    hospital = json.loads((ROOT / "data" / "hospital.json").read_text(encoding="utf-8"))
    (ROOT / "data" / "hospital.embed.js").write_text(
        "window.__HOSPITAL_BASE__=" + json.dumps(hospital, ensure_ascii=False) + ";\n",
        encoding="utf-8",
    )
    print("Wrote data/hospital.embed.js")

    for code in ("hy", "ru", "en"):
        data = json.loads((ROOT / "lang" / f"{code}.json").read_text(encoding="utf-8"))
        (ROOT / "lang" / f"{code}.embed.js").write_text(
            "window.__I18N__=window.__I18N__||{};"
            f"window.__I18N__['{code}']="
            + json.dumps(data, ensure_ascii=False)
            + ";\n",
            encoding="utf-8",
        )
        print(f"Wrote lang/{code}.embed.js")

    print("Done.")


if __name__ == "__main__":
    main()

File: scripts/build-embed.js
/**
 * Generates offline embed scripts so the site opens without a local server.
 * Run: node scripts/build-embed.js
 */
const fs = require('fs');
const path = require('path');

const root = path.join(__dirname, '..');

function writeEmbed(srcRelative, destRelative, globalName, nestedKey) {
  const src = path.join(root, srcRelative);
  const dest = path.join(root, destRelative);
  const data = JSON.parse(fs.readFileSync(src, 'utf8'));
  let js;
  if (nestedKey) {
    js = `window.${globalName} = window.${globalName} || {};\nwindow.${globalName}.${nestedKey} = ${JSON.stringify(data)};\n`;
  } else {
    js = `window.${globalName} = ${JSON.stringify(data)};\n`;
  }
  fs.writeFileSync(dest, js, 'utf8');
  console.log('Wrote', destRelative);
}

writeEmbed('data/hospital.json', 'data/hospital.embed.js', '__HOSPITAL_BASE__');
writeEmbed('data/about-article.json', 'data/about-article.embed.js', '__ABOUT_ARTICLE__');
['hy', 'ru', 'en'].forEach((code) => {
  writeEmbed(`lang/${code}.json`, `lang/${code}.embed.js`, '__I18N__', code);
});

console.log('Done. Site works via file:// and http://');
